src/storage/mvcc/txn.rs

```diff
@@ -350,6 +350,9 @@ impl<S: Snapshot> MvccTxn<S> {
         txn_size: u64,
         min_commit_ts: TimeStamp,
     ) -> Result<()> {
+        if mutation.should_not_write() {
+            return Ok(());
```

Comment:
Should we still clean the pessimistic lock here?

Reply:
for latest TiDB, we can ensure never send checkNotExists for pessimistic txn, maybe I should change `return` => `unreachable!`～？

Reply:
Got it, thanks!
`unreachable!` will make tikv panic. In case of TiDB changes the behavior, I think we can return `Err(box_err!("cannot handle checkNotExists in pessimistic prewrite"))`.